### PR TITLE
Feature/byzantine tests

### LINKED ISSUE #692: R&D: Conductor: Leader extends not notarized prev_block
Implement a test: Leader extends not notarized prev_block.
```
(T0) Leader_0: send Proposal0_0, extends not notarized prev_block
(T0) Leader_1: send Proposal1_0
(T0 + δ) Replica_0: reject Proposal0_, verify Proposal1_0
```

### LINKED ISSUE #697: Conductor: Protocol tells to verify all messages on the same rank
//Protocol tells to verify all messages on the same rank, alas we don’t follow protocol and vote for only one top-ranked block.
//Propose different block for every replica, this round should achieve notarization (with different block equality proof, not implemented yet)
```
(T0) Leader_0(Adv): send Proposal0_i to Replica_i i= notarization_threshold
(T0 + δ + Δ) Replica_i: send VerificationTicket0_i 
```

### LINKED ISSUE #698: Conductor: check no depletion, all first j leaders are adversarial
//check no depletion, all first j leaders are adversarial, they send different blocks to different replicas but not enough for notarization
```
(T0) Leader_0(Adv): send Proposalj_i to Replica_i, j= f , 0 < i < notarization_threshold
(T0 + δ + Δ) Replica_i: send VerificationTicket0_i 

```

### LINKED ISSUE #699: Conductor: check that proposer can’t break single block
//check that proposer can’t break single block 
```
(T0) Leader_0:  send Proposal0_0
(T0 + δ + Δ) Replica_i: send Verification_0
(T0 + 2δ + Δ) Leader_0: send Proposal0_1
(T0 + 3δ + Δ) Replica_i: send Notarization0_0 #ignore Proposal0_1

```

### LINKED ISSUE #701: Conductor: check make progress for an adversarial leader
//check make progress for an adversarial leader
```
(T0) Leader_0:  send Proposal0_0 for replica j , 0 <= j <1/2f
(T0) Leader_1:  send Proposal0_1
(T0 + δ + Δ) Replica_i: send Verification0_0 

```

### LINKED ISSUE #704: Conductor: Verify not existent block, this should work fine
//Verify not existent block, this should work fine, since we only formally verify verification_tickets
```
(T0) Leader_0: send Verification0_X
(T0 + δ) Replica_0: verify and forget

```

### LINKED ISSUE #707: Conductor: Half of nodes are down, generator proposes but can't get enough verification tickets
//Half of nodes are down, generator proposes but can't get enough verification tickets, restarts. Soon half of nodes get up and restarts round.
```
(T0) Replica_i: down, 0<=i<1/2f
(T0) Leader0_0: send Proposal0_0
(T0 + δ + Δ) Replica_0: send VerificationTicket0_0
(T0 + timeout) Replica_j: send VRFShare(timeout), 1/2f<j<=f
(T0 + 4*timeout) Replica_i: get up, send VRFShare(timeout), 0<=i<1/2f
(T0 + 4*timeout + δ): Leader0_1: send Proposal0_1
```

### LINKED ISSUE #708: Conductor: Malicious replicas start to send VRFShares for current round as if they get up after restart
//Malicious replicas start to send VRFShares for the current round as if they get up after restart
```
(T0) Leader_0: send Proposal0_0
(T0 + δ) Replica_i(Adv): send VRFShare(timeout), 0<=i<1/3f
(T0 + δ + Δ) Replica_j: send VerificationTicket0_j 
```

### LINKED ISSUE #709: Conductor: Miner sends previously proposed block used
//Miner sends previously proposed block used in the notarized round with one/several blocks notarized in it to propose in a new round
```
(T0) Leader_0: send Proposal0_0, Proposal0_1
(T0 + δ + Δ) Replica_j: send VerificationTicket0_j, VerificationTicket1_j  
(T0 + 2δ + Δ) Replica_j: send Notarization0_0, Notarization0_1 
(T0 + 3δ + Δ) Leader_1: send Proposal0_0
(T0 + 4δ + Δ) Replica_j: reject Proposal0_0
```

### LINKED ISSUE #817: Conductor: Notarize not existent block and fetch it
```
// Notarize not existent block and fetch it
(T0) Replica_i: send Verification_i_X  0<i<=2f/3 + 1
(T0 + δ) Replica_1: fetch Block_i
(T0 + 2*δ) Replica_1: send Notarization_i
```

### PR DESCRIPTION
**Pull request by MurashovVen**
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master BEFORE merging this PR  
  
Associated PRs (Link as appropriate):
...

### CODE REVIEW

**Review by dabasov — DISMISSED** [review]
lgtm

**Review by dabasov — DISMISSED** [review]
lgtm

**Review by dabasov — DISMISSED** [review]
lgtm

File: code/go/0chain.net/chaincore/chain/protocol_round_integration_tests.go [hunk 1]
```diff
@@ -88,6 +88,7 @@ func AddRoundInfoResult(r round.RoundI, finalisedBlockHash string) error {
 	res := roundInfo(r.GetRoundNumber(), finalisedBlockHash)
 	blob, err := res.Encode()
 	if err != nil {
+		log.Printf("1!")
```
Comment (dabasov) [inline code/go/0chain.net/chaincore/chain/protocol_round_integration_tests.go]: can we delete this log?

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | blobber-stake-nonce  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**